### PR TITLE
feat: func env vars and secrets docs

### PR DESCRIPTION
**Pull request by edwardfoyle**
_Issue #, if available:_
https://github.com/aws-amplify/amplify-cli/pull/7529
_Description of changes:_
Adds docs for upcoming release of Function secrets and environment variables. Draft because these changes shouldn't be merged until the release.
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/cli/function/env-vars.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+title: Custom environment variables
```
Comment (renebrandel) [inline docs/cli/function/env-vars.md:2]:
```suggestion
title: Environment variables
```

File: docs/cli/function/env-vars.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+description: Configure Lambda runtime environment variables
```
Comment (renebrandel) [inline docs/cli/function/env-vars.md:3]:
```suggestion
description: Configure environment variables for AWS Lambda functions
```

File: docs/cli/function/env-vars.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+---
+
+Amplify CLI allows you to create custom Lambda environment variables that will be available to your Lambda during runtime. Each Amplify environment can have a different value for an environment variable allowing use cases such as switching between dev and prod URLs depending on the environment.
```
Comment (renebrandel) [inline docs/cli/function/env-vars.md:6]:
```suggestion
Amplify CLI allows you to configure environment variables for your Lambda functions. Each Amplify environment can have a different value for an environment variable. For example, this allows you to switch between dev and prod URLs depending on the environment.
```

File: docs/cli/function/env-vars.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+> Environment variables should NOT be used for storing sensitive configuration values such as database passwords, API keys or access tokens. To access these types of values from your function, see [function secrets configuration](~/cli/function/secrets.md).
```
Comment (renebrandel) [inline docs/cli/function/env-vars.md:8]:
```suggestion
> Environment variables should NOT be used for storing sensitive configuration values such as database passwords, API keys or access tokens. Use [function secrets configuration](~/cli/function/secrets.md) instead!
```

File: docs/cli/function/env-vars.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+## Configuring environment variables
+To add a new function with custom environment variables, run `amplify add function`, select `yes` to the advanced settings prompt and select `yes` to the environment variables configuration prompt. From there, you will be able to specify a key and value for the environment variable.
```
Comment (renebrandel) [inline docs/cli/function/env-vars.md:11]:
```suggestion
To configure a new function with environment variables, run `amplify add function`, select `yes` to the advanced settings prompt and select `yes` to the environment variables configuration prompt. From there, you will be able to specify a key and value for the environment variable.
```

Comment (renebrandel) [reply docs/cli/function/env-vars.md:11]: also add the code flow maybe?

File: docs/cli/function/env-vars.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+To add custom environment variables to an existing function, run `amplify update function`, and select `Environment variables configuration`. You can then add, update and remove environment variables.
```
Comment (renebrandel) [inline docs/cli/function/env-vars.md:13]:
```suggestion
To configure environment variables for an existing function, run `amplify update function`, and select `Environment variables configuration`. You can then add, update, or remove environment variables.
```

Comment (renebrandel) [reply docs/cli/function/env-vars.md:13]: add code flow, maybe?

File: docs/cli/function/env-vars.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+## Multi-environment flows
+When creating a new Amplify environment using `amplify env add`, Amplify CLI will detect if some functions have environment variables. A prompt will appear asking if you want to apply all environment variable values to the new environment or make edits. If you choose to apply the existing values, you will still be able to make edits anytime using `amplify update function`.
```
Comment (renebrandel) [inline docs/cli/function/env-vars.md:16]:
```suggestion
When creating a new Amplify environment using `amplify env add`, Amplify CLI asks if you want to apply all environment variable values to the new environment or modify them. If you choose to apply the existing values, you can still make edits anytime by running `amplify update function`.
```

File: docs/cli/function/env-vars.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+When creating a new Amplify environment using `amplify env add --yes`, Amplify CLI will apply all environment variable values from the current environment to the new environment by default.
```
Comment (renebrandel) [inline docs/cli/function/env-vars.md:18]:
```suggestion
When creating a new Amplify environment using `amplify env add --yes`, Amplify CLI will apply all environment variable values from the current environment to the new environment.
```

File: docs/cli/function/secrets.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+> Add a secret
+  Update a secret
+  Remove secrets
+  I'm done
+```
+
+> Note: Amplify CLI never stores secrets locally. All secret values are immediately stored in [AWS Parameter Store](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html) using the SecureString parameter type.
+
+## Accessing the values in your function
+To access the secret values in your lambda function, use the [AWS SSM GetParameter API](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_GetParameter.html). Amplify CLI will automatically supply the SSM parameter name of the secret as an environment variable to the function. This value can be passed into the API call as the "Name" to retrieve the value. Ensure that the API call has "WithDecryption" specified as `true`.
```
Comment (renebrandel) [inline docs/cli/function/secrets.md:49]:
```suggestion
To access the secret values in your Lambda function, use the [AWS SSM GetParameter API](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_GetParameter.html). Amplify CLI will automatically supply the SSM parameter name of the secret as an environment variable to the function. This value can be passed into the API call as the "Name" to retrieve the value. Ensure that the API call has "WithDecryption" specified as `true`.
```

File: docs/cli/function/secrets.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+If your Lambda function is using the NodeJS runtime, a comment block will be placed at the top of your `index.js` file with example code to retrieve the secret values.
```
Comment (renebrandel) [inline docs/cli/function/secrets.md:51]: Maybe good to add a code snippet here of the comment block?

File: docs/cli/function/secrets.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+  Remove secrets
+  I'm done
+```
+
+> Note: Amplify CLI never stores secrets locally. All secret values are immediately stored in [AWS Parameter Store](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html) using the SecureString parameter type.
+
+## Accessing the values in your function
+To access the secret values in your Lambda function, use the [AWS SSM GetParameter API](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_GetParameter.html). Amplify CLI will automatically supply the SSM parameter name of the secret as an environment variable to the function. This value can be passed into the API call as the "Name" to retrieve the value. Ensure that the API call has "WithDecryption" specified as `true`.
+
+If your Lambda function is using the NodeJS runtime, a comment block will be placed at the top of your `index.js` file with example code to retrieve the secret values.
```
Comment (cjihrig) [inline docs/cli/function/secrets.md:51]:
```suggestion
If your Lambda function is using the Node.js runtime, a comment block will be placed at the top of your `index.js` file with example code to retrieve the secret values.
```

**Review by cjihrig — APPROVED** [review]

**Review by mauerbac — APPROVED** [review]
lgtm

**Pull request merged**